packages/vendure-plugin-accept-blue/src/api/accept-blue-service.ts

```diff
@@ -68,31 +79,186 @@ export class AcceptBlueService {
+        id: order.customer?.id,
+        customFields: { activeBlueCustomerId: customer.id },
+      });
+    }
     // Save payment method
-    const paymentMethod = await client.getOrCreatePaymentMethod(
+    const paymentMethod = await client.getOrCreateCardPaymentMethod(
       customer.id,
       ccDetails
     );
-    return await this.payWithSavedPaymentMethod(
+    return await this.payWithCardPaymentMethodDetails(
+      ctx,
+      order,
+      amount,
+      client,
+      paymentMethod.id,
+      ccDetails
+    );
+  }
+
+  async payWithToken(
+    ctx: RequestContext,
+    order: Order,
+    amount: number,
+    client: AcceptBlueClient,
+    ccDetails: TokenPaymentMethodInput
+  ): Promise<HandlePaymentResult> {
+    if (!order.customer) {
+      throw new UserInputError(`Order must have a customer`);
+    }
+    const customer = await client.getOrCreateCustomer(
+      order.customer.emailAddress
+    );
+    if (!order.customer?.customFields?.activeBlueCustomerId) {
+      await this.customerService.update(ctx, {
+        id: order.customer?.id,
+        customFields: { activeBlueCustomerId: customer.id },
+      });
+    }
+    // Save payment method
+    const methods = await client.getPaymentMethods(
+      order.customer.customFields.activeBlueCustomerId
+    );
+    const paymentMethod = methods.find((method) =>
+      isSameCard(ccDetails, method as AcceptBlueCardPaymentMethod)
+    );
+    if (!paymentMethod) {
+      throw new UserInputError(
+        'No Saved Credit Card with this toknized nonce code'
+      );
+    }
+    return await this.payWithCardPaymentMethodDetails(
       ctx,
       order,
       amount,
       client,
-      paymentMethod.id
+      paymentMethod.id,
+      ccDetails
     );
   }
 
+  async payWithCheck(
+    ctx: RequestContext,
+    order: Order,
+    amount: number,
+    client: AcceptBlueClient,
+    ccDetails: CheckPaymentInput
+  ): Promise<HandlePaymentResult> {
+    if (!order.customer) {
+      throw new UserInputError(`Order must have a customer`);
+    }
+    const customer = await client.getOrCreateCustomer(
+      order.customer.emailAddress
+    );
+    if (!order.customer?.customFields?.activeBlueCustomerId) {
+      await this.customerService.update(ctx, {
+        id: order.customer?.id,
+        customFields: { activeBlueCustomerId: customer.id },
+      });
+    }
+    // Save payment method
+    const paymentMethod = await client.getOrCreateCheckPaymentMethod(
+      customer.id,
+      ccDetails
+    );
+    return await this.payWithCheckPaymentMethodDetails(
+      ctx,
+      order,
+      amount,
+      client,
+      paymentMethod.id,
+      ccDetails
+    );
+  }
+
+  async payWithCardPaymentMethodDetails(
+    ctx: RequestContext,
+    order: Order,
+    amountDueNow: number,
+    client: AcceptBlueClient,
+    paymentMethodId: number,
+    ccDetails: CreditCardPaymentInput | TokenPaymentMethodInput
+  ): Promise<HandlePaymentResult> {
+    const recurringSchedules =
```

Comment:
I see that you are only passing the `paymentMethodId` in here to create the recurring schedules here. This function now does 2 things: create recurring schedules and create a one time payment.

To avoid too many function layers (too many function to function to function calls), it's better to use this flow in the upper `payWithCreditCard()` function:

```ts
await createRecurringSchedule(ctx, order, paymentMethod);
// Preferably use paymentMethodId, if not possible pass ccDetails
await createOneTimePayment(ctx, order, amountDueNow, paymentMethodId)
```
This way you have 2 clear separate functions `createRecurringSchedule` and `createOneTimePayment`, instead of `payWithCardPaymentMethodDetails` and `payWithCreditCard` (they sound pretty similar right?)

It's usually a sign things can be structured better if you are passing the same arguments (paymentMethodId, client) down multiple layers of functions.